src/GenericTreeNode.php

```diff
@@ -76,6 +80,11 @@ public function getParent(): ?TreeNode
     public function setParent(?TreeNode $parent): void
     {
         $this->parent = $parent;
+        if ($parent instanceof TreeNode) {

```

Comment:
$rootId = (null === $parent) ? $this->getId() : $parent->getRootId();
    $this->setRootId($rootId);
    
Parent kann nur null oder TreeNode sein - also müssen wir nicht (aufwändiger) prüfen ob es TreeNode ist - nur, dass es nicht-null ist. :)

Reply:
Done